### ISSUE TITLE
Add ability to scale `Shots` object with `__mul__` and `*`

### ISSUE THREAD

**Issue by albi3ro** [PennyLaneAI/pennylane#4910]
PennyLane has a `qml.measurements.Shots` object that stores information about the number of shots:

```pycon
>>> qml.measurements.Shots(100)
Shots(total_shots=100, shot_vector=(ShotCopies(100 shots x 1),))
>>> qml.measurements.Shots((100,100))
Shots(total_shots=200, shot_vector=(ShotCopies(100 shots x 2),))
```

This class is located at [`pennylane/measurements/shots.py`](https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/measurements/shots.py)

This task is to define ` __mul__` method to allow scaling the total number of shots. If the shots class has "shot vector", each component of the shot vector should be scaled as well. If the resulting number of shots is not an integer, we should round down.

This should look like:

```pycon
>>> 2 * Shots(100)
Shots(total_shots=200, shot_vector=(ShotCopies(200 shots x 1),))
>>> 2 * Shots((100, 100))
Shots(total_shots=400, shot_vector=(ShotCopies(200 shots x 2),))
>>> 0.5 * Shots(5)
Shots(total_shots=2, shot_vector=(ShotCopies(2 shots x 1),))
```

 A PR solving this issue would:

* Include a changelog entry at `docs/releases/changelog-dev.md`
* Include tests in [`tests/measurements/test_shots.py`](https://github.com/PennyLaneAI/pennylane/blob/master/tests/measurements/test_shots.py)
* Pass required pylint and black checks

**Comment by AnuravModak** [comment]
if its available for external contributors let me know!

**Comment by albi3ro** [comment]
@AnuravModak This issue is available for external contributors. You are welcome to work on it.

**Comment by AnuravModak** [comment]
Now for this multiplication operation is it commutative in nature, For instance will --> 2 * Shots(100)==Shots(100)*2?

Basically i want to confirm, do i also need to implement ` __rmul__ (self, scalar)` with this or not? If yes, is the multiplication operation for shots holds the commutative property of multiplication?

Mathematically, for any numbers "a" and "b"
a×b=b×a.    ----> commutative property

please advise.

**Comment by albi3ro** [comment]
Thanks for the question. Yes it should be commutative, so right and left multiplication should be the same. 

I also forgot to clarify this. "Analytic" with `Shots(None)` should remain `Shots(None)` no matter what it is multiplied by.

**Comment by albi3ro** [comment]
Closed by #4913
